src/main/java/org/jabref/logic/citationstyle/CitationStyle.java

```diff
@@ -153,8 +154,10 @@ public static CitationStyle getDefault() {
 
             try (FileSystem jarFs = FileSystems.newFileSystem(Paths.get(path), null)) {
 
-                List<Path> allStyles = Files.find(jarFs.getRootDirectories().iterator().next(), 1, (file, attr) -> file.toString().endsWith("csl")).collect(Collectors.toList());
-
+                List<Path> allStyles;
+                try (Stream<Path> stylefileStream = Files.find(jarFs.getRootDirectories().iterator().next(), 1, (file, attr) -> file.toString().endsWith("csl"))) {
```

Comment:
This is again one of the ugly cases where an unchecked exception could be thrown. Because streams and lambdas can't throw checked exceptions. In this case it could be that the stream is already closed when calling collect. 
 Similar to #3606

Reply:
Yes, please catch `IOUncheckedException` and rethrow it as `IOException`.